### PR TITLE
ticket 11 completed

### PR DESCRIPTION
**Pull request by Amir2341**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -215,6 +215,80 @@ describe("GET /api/articles", () => {
         );
       });
   });
+  test("status 200 return article array sorted by author ", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js]: This description should be more generic please! It's not specifically testing for sorted_by author, it's just that there's a sort_by in general. We're just using author in this case to prove that it works

File: __tests__/app.test.js [hunk 2]
```diff
@@ -215,6 +215,80 @@ describe("GET /api/articles", () => {
...
+              topic: expect.any(String),
+              author: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+            }),
+          ])
+        );
+      });
+  });
+  test("status 200 return article array sorted by created_at and order desc with topic cats", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js]: Same as above for all of these please! We're testing the general behaviour - it won't always be cats.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -215,6 +215,80 @@ describe("GET /api/articles", () => {
...
+              topic: "cats",
+              author: "rogersop",
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+            }),
+          ])
+        );
+      });
+  });
+  test("status 404 invalid query", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js]: Please test for each invalid query - e.g. invalid sort_by and order_by. Also please test for a situation in which the topic exists but there are no associated articles.

File: controllers/controller.js [hunk 4]
```diff
@@ -45,10 +45,17 @@ exports.getUsers = (req, res) => {
   });
 };
 
-exports.getAllArticles = (req, res) => {
-  selectAllArticles().then((articles) => {
-    res.send({ articles });
-  });
+exports.getAllArticles = (req, res, next) => {
+  const { sort_by } = req.query;
```
Comment (Rosem10) [inline controllers/controller.js]: You could do this all on one like like {sort_by, order, topic} = req.query

**Review by Rosem10 — CHANGES_REQUESTED** [review]
Couple more tests to add please!

**Review by HeatherEver — APPROVED** [review]

**Pull request merged**